lib/module.js

```diff
@@ -88,9 +88,18 @@ function readPackage(requestPath) {
 function tryPackage(requestPath, exts) {
   var pkg = readPackage(requestPath);
 
-  if (!pkg) return false;
+  if (!(pkg && pkg.main)) return false;
+
+  var resolved = pkg['_resolved'],
```

Comment:
I see.  What if _resolved is not as unique as this patch assumes?  Things will break badly, won't they?